### ISSUE TITLE
Overflow issue with `logdet` function for computing the diversity

### ISSUE THREAD

**Issue by FanwangM** [theochem/Selector#229]
When computing with diversity with the `logdet` function, the current implementation can lead to numerical instability issues.

```python
import numpy as np
# numpy version, 1.26.4

# the mid + np.identity(len(X)) part in the original code
data = np.loadtxt("test.txt")
data = data["arr_01"]

np.log10(np.linalg.det(data))
# ...[python3.11/site-packages/numpy/linalg/linalg.py:2180](.../lib/python3.11/site-packages/numpy/linalg/linalg.py:2180): RuntimeWarning: overflow encountered in det
#  r = _umath_linalg.det(a, signature=signature)

```

This should be fixed. @FanwangM 

[test.txt](https://github.com/user-attachments/files/16103102/test.txt)

**Comment by FanwangM** [comment]
This can be fixed with https://github.com/numpy/numpy/blob/b46446e58386beb3b55633421cea069fb1a555dd/numpy/linalg/_linalg.py#L2286.

**Comment by FanwangM** [comment]
Fixed in #230.
